felix/wireguard/bootstrap.go

```diff
@@ -0,0 +1,70 @@
+package wireguard
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+
+	"github.com/projectcalico/calico/felix/netlinkshim"
+	"github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+	cerrors "github.com/projectcalico/calico/libcalico-go/lib/errors"
+	"github.com/projectcalico/calico/libcalico-go/lib/options"
+)
+
+// BootstrapHostConnectivity forces WireGuard peers with hostencryption enabled to communicate with this node unencrypted.
+// This ensures connectivity in scenarios where we have lost our WireGuard config, but will be sent WireGuard traffic
+// e.g. after a node restart, during felix startup, when we need to fetch config from Typha (calico/issues/5125)
+func BootstrapHostConnectivity(wgDeviceName string, nodeName string, calicoClient clientv3.Interface) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	thisNode, err := calicoClient.Nodes().Get(ctx, nodeName, options.GetOptions{})
+	cancel()
+	if err != nil {
+		log.WithError(err).Warn("could not fetch node config from datastore")
```

Comment:
Logs should be in `Sentence case` They're meant to be read by the end user.

Reply:
Fixed up the logs now, I assume that error strings don't count? I've left out any errors made with `fmt.Errorf` or similar for now